modal/cls.py

```diff
@@ -475,11 +475,10 @@ def validate_construction_mechanism(user_cls):
         annotated_params = {k: t for k, t in annotations.items() if k in params}
         for k, t in annotated_params.items():
             if t not in CLASS_PARAM_TYPE_MAP:
-                t_name = getattr(t, "__name__", repr(t))
-                supported = ", ".join(t.__name__ for t in CLASS_PARAM_TYPE_MAP.keys())
-                raise InvalidError(
-                    f"{user_cls.__name__}.{k}: {t_name} is not a supported parameter type. Use one of: {supported}"
-                )
+                pass
+                # TODO:
+                # raise if cls has webhooks
+                # and no default value for pickle parameter
```

Comment:
I guess the previous check is no longer necessary since the new type will be "catch-all" -- however we could add that too if we add an explicit type that a user must use instead

Reply:
nvm we can keep this and require to use the correct annotation